cmd/snap/cmd_auto_import_test.go

```diff
@@ -321,3 +323,111 @@ func (s *SnapSuite) TestAutoImportUnhappyInInstallMode(c *C) {
 	c.Check(s.Stdout(), Equals, "")
 	c.Check(s.Stderr(), Equals, "auto-import is disabled in install-mode\n")
 }
+
+var mountStatic = []string{"mount", "-t", "ext4,vfat", "-o", "ro", "--make-private"}
+
+func (s *SnapSuite) TestAutoImportFromRemovable(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	_, restoreLogger := logger.MockLogger()
+	defer restoreLogger()
+
+	rootdir := c.MkDir()
+	dirs.SetRootDir(rootdir)
+
+	var umounts []string
+	restore = snap.MockSyscallUmount(func(p string, _ int) error {
+		umounts = append(umounts, p)
+		return nil
+	})
+	defer restore()
+
+	var tmpdirIdx int
+	restore = snap.MockIoutilTempDir(func(where string, p string) (string, error) {
+		c.Check(where, Equals, "")
+		tmpdirIdx++
+		return filepath.Join(rootdir, fmt.Sprintf("/tmp/%s%d", p, tmpdirIdx)), nil
+	})
+	defer restore()
+
+	mountCmd := testutil.MockCommand(c, "mount", "")
+	defer mountCmd.Restore()
+
+	snaptest.PopulateDir(rootdir, [][]string{
+		// removable without partitions
+		{"sys/block/sdremovable/removable", "1\n"},
+		// fixed disk
+		{"sys/block/sdfixed/removable", "0\n"},
+		// removable with partitions
+		{"sys/block/sdpart/removable", "1\n"},
+		{"sys/block/sdpart/sdpart1/partition", "1\n"},
+		{"sys/block/sdpart/sdpart2/partition", "0\n"},
+		{"sys/block/sdpart/sdpart3/partition", "1\n"},
+		// removable but subdevices are not partitions?
+		{"sys/block/sdother/removable", "1\n"},
+		{"sys/block/sdother/sdother1/partition", "0\n"},
+	})
+
+	// do not mock mountinfo contents, we just want to observe whether we
+	// try to mount and umount the right stuff
+
+	_, err := snap.Parser(snap.Client()).ParseArgs([]string{"auto-import"})
+	c.Assert(err, IsNil)
+	c.Check(s.Stdout(), Equals, "")
+	c.Check(s.Stderr(), Equals, "")
+	c.Check(mountCmd.Calls(), DeepEquals, [][]string{
```

Comment:
Glob() does not sort, so either you should sort the output from Glob(), or sort the output here. I have a slight preference to sort the Glob() output in the code.

Reply:
Fair enough, it's an implementation detail of Glob that's not mentioned in the docs, so making it explicit makes sense.